### DISCUSSION TITLE
Spacegroup changed in the relaxation

### DISCUSSION THREAD

**Discussion by kw600** [SSCHAcode/python-sscha#201] [Q&A] [answered]
Dear SSCHA developers,

I am using the same size supercell to be commensurate with 3 phases in perovskite (cubic, tetragonal, orth).  So I am thinking to use the supercell and only calculate its Gamma point phonon. I do not why it seems like the space group is changed from the inital P4/mbm to P1 after relaxation. I try to use spglib.get_spacegroup to detect the final space group and I need a tolerance of 0.2 angstrom to go back to P4/mbm.  

Thanks for any help!
Kang

**Comment by mesonepigreco** [comment, marked as answer]
Hi,
It is possible that the symmetry engine does not recognize correctly the symmetries in a supercell.
You can switch to a more clever engine like spglib by setting:
```python
minim.use_spglib = True
```
where ``minim`` is the ``SSCHA_Minimizer`` object.
Indeed, the structure of the starting dynamical matrix must satisfy the symmetry group you want to constrain.
To check it, use
```
print(spglib.get_spacegroup(dyn.structure.get_ase_atoms()))
```
where ``dyn`` is your starting dynamical matrix. If it recognizes the correct group, the use_spglib flag should ensure that the symmetry group is preserved during the minimization.

I hope this was helpful,
Best regards,
Lorenzo

**Reply by kw600** [reply]
Dear Lorenzo,

Thank you very much for your help. The use of spglib has fixed my issue. The space group can be correctly maintained.

Best wishes,
Kang

**Comment by Kurufinve** [comment]
Dear Lorenzo, 

Can such result (changing the symmetry group from P4/mbm to P1) indicate that P4/mbm structure is dynamically unstable at given conditions and the SSCHA relaxation (that does not recognize the symmetries in supercell during minimization) simply produce a dynamically stable structure that has lower symmetry? 

Sincerely,
Daniil Poletaev

**Reply by mesonepigreco** [reply]
I would not say so, 
Remember that you have stochastic sampling; therefore, your averages have a stochastic error. If the symmetries are not imposed, you always have a P1 symmetry group because of stochastic noise. 
In particular, P1 means no symmetries at all, which is very rare and most likely an indication of just some stochastic noise around a high-symmetry structure.

**Reply by Kurufinve** [reply]
Dear Lorenzo, thank you very much for such fast reply! :) 
This my question is a consequence of more common question about stabilizing the structure after SSCHA relaxation (from here: https://github.com/orgs/SSCHAcode/discussions/175). Performing SSCHA relaxations for different structures I found that dynamically unstable structure with lower symmetry can be transformed to the dynamically stable structure with higher symmetry after SSCHA relaxation (like R-3m to Fm-3m in the example with LaH10) but not vice verca. For example, hexagonal YH9 that is dynamically unstable in harmonic approximation at 200 GPa does not change its symmetry group after the SSCHA relaxation at this pressure (fixed cell + variable cell relaxation with fixed volume), however the calculated anharmonic Hessian also indicates that it is unstable (I expected that SSCHA relaxation will change its symmetry group to lower one and make it dynamically stable). I supposed that this happened due to symmetry restrictions imposed on auxiliary dynamical matrix within the minimization procedure. Am I right? Or it should not make influence on the shape of final structure after SSCHA relaxation? 
The main concern is: can we tune the SSCHA minimization and relaxation settings in such a way that allow us to transform the dynamically unstable structure to dynamically stable one after the SSCHA relaxation?
